math/mathcore/inc/Math/IFunction.h

```diff
@@ -232,6 +240,14 @@ namespace ROOT {
             function to evaluate the derivative with respect each coordinate. To be implemented by the derived class
          */
          virtual T DoDerivative(const T *x, unsigned int icoord) const = 0;
+         /// In some cases, the derivative algorithm will use information from the previous step, these can be passed
+         /// in with this overload. The `previous_*` arrays can also be used to return second derivative and step size
+         /// so that these can be passed forward again as well at the call site, if necessary.
+         virtual T DoDerivative(const T *x, unsigned int icoord, T * /*previous_grad*/, T * /*previous_g2*/,
+                                T * /*previous_gstep*/) const
```

Comment:
```suggestion
         virtual T DoDerivativeWithPreviousResult(const T *x, unsigned int icoord, T * /*previous_grad*/, T * /*previous_g2*/,
                                T * /*previous_gstep*/) const
```
What about giving these new functions a different name? There is no reason to also name it `DoDerivative`, and like this we avoid the compiler warning when only the old `DoDerivative` signature is overridden.